features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/TimelineView.kt

```diff
@@ -101,12 +72,25 @@ fun TimelineView(
     onMessageClicked: (TimelineItem.Event) -> Unit = {},
     onMessageLongClicked: (TimelineItem.Event) -> Unit = {},
 ) {
+    val coroutineScope = rememberCoroutineScope()
 
     fun onReachedLoadMore() {
         state.eventSink(TimelineEvents.LoadMore)
     }
 
     val lazyListState = rememberLazyListState()
+
+    fun inReplyToClicked(eventId: EventId) {
+        val index = state.timelineItems.indexOfFirst { it.identifier() == eventId.value }
+        if (index >= 0) {
+            coroutineScope.launch {
+                lazyListState.animateScrollToItem(index)
+            }
+        } else {
+            // TODO should we paginate first?
```

Comment:
I'm not sure about what we should do here, to be honest. Should we block the UI until we've paginated back to that item? What if we somehow reference a non-existing one?

Reply:
We should be able to call the context API somehow with the rust-sdk, so it'll fail if the event doesn't exist (which should not happen I guess?)

Reply:
We're having a kick off meeting with iOS and design today so I'll ask about how this should work both UX and logic wise.

Reply:
Apparently we're removing this logic altogether for now since we can't just back-paginate indefinitely.

Reply:
But like I said, the `context` api should return the event and some of the prev and next events. It's what is used for permalink too. But probably not available yet in the rust sdk?

Reply:
Yes, I think SlidingSync can only handle back pagination right now and this would be like jumping in time. We'd probably have to add some kind of "load next/load previous" marks between the latest items and those retrieved with the API.